bindings/matrix-sdk-ffi/src/encryption.rs

```diff
@@ -410,6 +410,63 @@ impl Encryption {
     pub async fn wait_for_e2ee_initialization_tasks(&self) {
         self.inner.wait_for_e2ee_initialization_tasks().await;
     }
+
+    /// Get the E2EE identity of a user.
+    ///
+    /// Returns an error if this user does not exist, if there is an error
+    /// contacting the crypto store, or if our client is not logged in.
+    pub async fn get_user_identity(
+        &self,
+        user_id: String,
+    ) -> Result<Arc<UserIdentity>, ClientError> {
+        Ok(Arc::new(UserIdentity {
+            inner: self
+                .inner
+                .get_user_identity(user_id.as_str().try_into()?)
+                .await?
+                .ok_or(ClientError::new("User not found"))?,
+        }))
+    }
+}
+
+/// The E2EE identity of a user.
+#[derive(uniffi::Object)]
+pub struct UserIdentity {
+    inner: matrix_sdk::encryption::identities::UserIdentity,
+}
+
+#[uniffi::export]
+impl UserIdentity {
+    /// Remember this identity, ensuring it does not result in a pin violation.
+    ///
+    /// When we first see a user, we assume their cryptographic identity has not
+    /// been tampered with by the homeserver or another entity with
+    /// man-in-the-middle capabilities. We remember this identity and call this
+    /// action "pinning".
+    ///
+    /// If the identity presented for the user changes later on, the newly
+    /// presented identity is considered to be in "pin violation". This
+    /// method explicitly accepts the new identity, allowing it to replace
+    /// the previously pinned one and bringing it out of pin violation.
+    ///
+    /// An identity which is not verified and is in pin violation will require
+    /// user approval before it can be used.
+    pub(crate) async fn pin(&self) -> Result<(), ClientError> {
+        Ok(self.inner.pin().await?)
+    }
+
+    /// Get the public part of the Master key of this user identity.
+    ///
+    /// The public part of the Master key is usually used to uniquely identify
+    /// the identity.
+    pub(crate) fn master_key(&self) -> Result<String, ClientError> {
```

Comment:
This is fine from my point of view, though usually we return an `Result<Option<T>>` for these cases since it's easier to distinguish the "Missing master key" case from other error types. This is especially true for `ClientError` which has only a single variant and people would need to match on the string.

The same applies to the `get_user_identity()` method.

Reply:
Fixed in 562316c4e5